src/compiler/checker.ts

```diff
@@ -30690,6 +30682,11 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return jsxNamespace && getSymbol(jsxNamespace.exports!, JsxNames.LibraryManagedAttributes, SymbolFlags.Type);
     }
 
+    function getJsxElementTypeSymbol(jsxNamespace: Symbol) {
+        // JSX.ElementType [symbol]
+        return jsxNamespace && getSymbol(jsxNamespace.exports!, JsxNames.ElementType, SymbolFlags.Type);

```

Comment:
probably don't need to check jsxNamespace when its type doesn't include undefined

Reply:
`getJsxNamespaceAt` quite literally has a `return undefined!; // TODO: GH#18217` in it, and we check its' results for defined-ness in _most_ other places because of this. It just has inaccurate types and I'm not really sure why. Old API compat?